elytron/WFCORE-4360-CredentialStore_Expression_Resolution.adoc

```diff
@@ -0,0 +1,420 @@
+more `AES` `SecretKey` instances.  This enhancement will provide utilities to encrypt values using the keys in the credential stores, these can then be placed
+into the WildFly management model on any resource that supports expressions.  At runtime a new expression resolver resource will be available within WildFly to
+dynamically decrypt the expressions by also making use of the `SecretKey` instances loaded from one or more credential stores.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-4360[WFCORE-4360] - Support encrypted expression resolution using a CredentialStore.
+* https://issues.jboss.org/browse/WFLY-12215[WFLY-12215] - Support encrypted expression resolution using a CredentialStore.
+* https://issues.jboss.org/browse/ELY-1842[ELY-1842] - Command line support for `SecretKey` manipulation.
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-677[EAP7-677]
+* https://issues.redhat.com/browse/WFCORE-4584[WFCORE-4584] - Include the new wildfly-elytron-encryption module.
+* https://issues.redhat.com/browse/ELY-2055[ELY-2055] - Refactor CredentialStoreCommand to make development easier.
+* https://issues.redhat.com/browse/ELY-1850[ELY-1850] - CredentialStore API missing ability to check alias types.
+* https://issues.redhat.com/browse/ELY-2081[ELY-2081] - Develop a simple properties based credential store for secret keys.
+* https://issues.redhat.com/browse/ELY-2086[ELY-2086] - Update the WildFly Elytron tool to support encrypting clear text values.
+
+=== Other Links
+
+The following is a link to an early discussion on the mailto:wildfly-dev@lists.jboss.org[wildfly-dev] mailing list.
+
+https://lists.jboss.org/pipermail/wildfly-dev/2019-July/006932.html[Expression Resolver Discussion]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:okotek@redhat.com[Ondrej Kotek]
+
+=== Testing By
+
+[ ] Engineering
+
+[x] QE
+
+=== Affected Projects or Components
+
+Within the WildFly Elytron project new utilities will be added to handle the generation of `SecretKey` instances and the encryption and decryption of clear
+text values.  These utilities will support transfer of the values using versioned Base64 encoded tokens which will be described in more detail below.
+
+Within WildFly Core the general `ExpressionResolver` support will be updated to detect the availability of an expression resolver capability exposing
+a runtime API for expression resolution.
+
+Also within WildFly Core the Elytron subsystem will be updated with new resources as required, resources will also contain operations both for key management
+and operations for the encryption of clear text strings for use as expressions in the management model.
+
+Additionally all documentation lives within the WildFly project so updates will be applied there accordingly.
+
+=== Other Interested Projects
+
```

Comment:
[Relevant Installation Types](https://github.com/wildfly/wildfly-proposals/blob/master/design-doc-template.adoc#relevant-installation-types) is missing.

Reply:
I will add this but there will be no exclusions, this will be a core feature applicable in all installation types.

Reply:
Done

Reply:
Thank you, this can be resolved.